File: fintechplatform/src/main/java/com/fintechplatform/android/profile/contacts/ui/ContactsUI.java
package com.fintechplatform.android.profile.contacts.ui;

import android.content.Context;
import android.content.Intent;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.android.api.NetModule;
import com.fintechplatform.android.models.DataAccount;
import com.fintechplatform.android.profile.api.ProfileAPIModule;

public class ContactsUI {
    private String hostName;
    private DataAccount configuration;

    static ContactsUI instance;

    public ContactsUI(String hostName, DataAccount configuration) {
        this.hostName = hostName;
        this.configuration = configuration;
    }

    private ContactsViewComponent buildContactsViewComponet(Context context, ContactsContract.View view) {
        return DaggerContactsViewComponent.builder()
                .netModule(new NetModule(Volley.newRequestQueue(context), instance.hostName))
                .contactsPresenterModule(new ContactsPresenterModule(view, instance.configuration))
                .profileAPIModule(new ProfileAPIModule(instance.hostName))
                .build();
    }
    public ContactsViewComponent createContactsComponent(Context context, ContactsContract.View view) {
        return instance.buildContactsViewComponet(context, view);
    }

    public void start(Context context) {
        instance = this;
        Intent intent = new Intent(context, ContactsActivity.class);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/contacts/ContactsUI.kt
package com.fintechplatform.ui.profile.contacts

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.profile.api.IdsDocumentsAPIModule
import com.fintechplatform.api.profile.api.ProfileAPIModule
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.contacts.di.ContactsPresenterModule
import com.fintechplatform.ui.profile.contacts.di.ContactsViewComponent
import com.fintechplatform.ui.profile.contacts.di.DaggerContactsViewComponent

class ContactsUI(private val hostName: String, private val configuration: DataAccount) {

    fun startActivity(context: Context) {
        val intent = Intent(context, ContactsActivity::class.java)
        val args = Bundle()
        args.putString("hostname", hostName)
        args.putParcelable("dataAccount", configuration)
        intent.putExtras(args)
        context.startActivity(intent)
    }

    fun creatFragment(hostName: String, dataAccount: DataAccount): ContactsFragment {
        return ContactsFragment.newInstance(hostName, dataAccount)
    }

    object Builder {
        fun buildContactsViewComponet(context: Context, view: ContactsContract.View, hostName: String, dataAccount: DataAccount): ContactsViewComponent {
            return DaggerContactsViewComponent.builder()
                    .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                    .contactsPresenterModule(ContactsPresenterModule(view, dataAccount))
                    .profileAPIModule(ProfileAPIModule(hostName))
                    .idsDocumentsAPIModule(IdsDocumentsAPIModule(hostName))
                    .build()
        }

    }
}
